addons/purchase/models/purchase_order_line.py

```diff
@@ -355,13 +358,24 @@ def _compute_price_unit_and_date_planned_and_name(self):
                     line.date_order or fields.Date.context_today(line),
                     False
                 )
+
+                is_computed_price = line.price_unit in (0, price_unit, line._origin.price_unit, line.product_uom._compute_price(price_unit, line.product_id.uom_po_id))
```

Comment:
Accessing `_origin` in a compute is generally a bad idea. If your compute is triggered through some python code (e.g. some code changed the `product_qty`), then `_origin` won't contain the old value by the time the value is actually recomputed.